app/src/androidTestGecko/java/org/mozilla/focus/activity/SearchEngineSelectionTest.java

```diff
@@ -0,0 +1,167 @@
+    public ActivityTestRule<MainActivity> mActivityTestRule
+            = new ActivityTestRule<MainActivity>(MainActivity.class) {
+
+        @Override
+        protected void beforeActivityLaunched() {
+            super.beforeActivityLaunched();
+
+            Context appContext = InstrumentationRegistry.getInstrumentation()
+                    .getTargetContext()
+                    .getApplicationContext();
+            PreferenceManager.getDefaultSharedPreferences(appContext)
+                    .edit()
+                    .putBoolean(FIRSTRUN_PREF, true)
+                    .apply();
+        }
+    };
+
+    @After
+    public void tearDown() throws Exception {
+        mActivityTestRule.getActivity().finishAndRemoveTask();
+    }
+
+    @Ignore("https://github.com/mozilla-mobile/focus-android/issues/2103")
+    @Test
+    public void SearchTest() throws InterruptedException, UiObjectNotFoundException {
+
+        UiObject SearchEngineSelection = TestHelper.settingsList.getChild(new UiSelector()
+                .className("android.widget.LinearLayout")
+                .instance(1));
+        UiObject searchName = SearchEngineSelection.getChild(new UiSelector()
+                .resourceId("android:id/title")
+                .enabled(true));
+
+        /* Go to Settings and select the Search Engine */
+        assertTrue(TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime));
+
+        openSettings();
+
+        /* Set the search engine to Google */
+        SearchEngineSelection.click();
+
+        /* Get the dynamically generated search engine list from this settings page */
+        UiScrollable SearchEngineList = new UiScrollable(new UiSelector()
+                .resourceId("org.mozilla.klar.gecko.debug:id/search_engine_group").enabled(true));
+
+        UiObject GoogleSelection = SearchEngineList.getChildByText(new UiSelector()
+                .className(RadioButton.class), "Google");
+        UiObject YahooSelection = SearchEngineList.getChildByText(new UiSelector()
+                .className(RadioButton.class), "Yahoo");
+
+        GoogleSelection.waitForExists(waitingTime);
+        GoogleSelection.click();
+        TestHelper.pressBackKey();
+        // Now it's changed to Google
+        assertTrue(searchName.getText().equals("Google"));
+        TestHelper.settingsHeading.waitForExists(waitingTime);
+        TestHelper.pressBackKey();
+
+        /* load blank spaces and press enter key for search, it should not do anything */
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.clearTextField();
+        TestHelper.inlineAutocompleteEditText.setText("   ");
+        TestHelper.pressEnterKey();
+        assertTrue(TestHelper.inlineAutocompleteEditText.exists());
+
+        /* Now do some search */
+        TestHelper.inlineAutocompleteEditText.clearTextField();
+        TestHelper.inlineAutocompleteEditText.setText("mozilla focus");
```

Comment:
\[PMD\] INFO: The String literal "mozilla focus" appears 4 times in this file; the first occurrence is on line 107